app/tests/containers/Organisation/Facility.test.tsx

```diff
@@ -40,12 +40,13 @@ describe('Facility', () => {
     expect(r.find('CardHeader').text()).toMatch(/Facility Name/);
   });
 
-  it('should not render a Button if an applicationID exists and is not in draft status', async () => {
+  it('should render a View Submitted App button if an applicationID exists and is not in draft status', async () => {
     const r = shallow(
       <FacilityComponent relay={{environment: {}}} facility={facility} />
     );
     expect(r).toMatchSnapshot();
-    expect(r.exists('Button')).toEqual(false);
+    expect(r.exists('Button')).toEqual(true);
+    expect(r.find('Button').prop('data-action')).toBe('view');
```

Comment:
if we bind and event handler instead, we can use enzyme to dispatch a synthetic event and test that code path

Reply:
something along the lines of...
```
const button = r.find('Button');
button.simulate('click', new Event('click'));
```

Reply:
Here I'm testing that the button is a `view` application button. I'm not testing events. Just to see that right button is showing up.

Reply:
I think that what @wenzowski  means is that instead of testing for the button text, or a `data-` attribute, you can test that clicking the button will trigger the correct handler (i.e. the "Start Application" button invokes `handleStartApplication`). In the case of the "view" button, I suppose you want to check the presence of a next.js `Link`

Reply:
```
const spy = jest.spyOn(wrapper.instance(), 'handleStartApplication'); // replace function
wrapper.update(); // not sure this is needed
const button = r.find('Button');
button.simulate('click');
expect(spy).toBeCalled(); 
```

Reply:
Exactly...with the caveat that `button.simulate('click');` may need an event passed in as well when you're shallow rendering with Enzyme.

Reply:
If the button is always contained in a `<Link><Button /><Link>` and that's where the click handler resides, maybe you're trying to assert that 
`r.find('Button').parent().is('Link')` and 
`r.find('Button').parent().prop('href').is({"pathname": "/view-application", "query": {"applicationId": "id"})`
...or something like that?

Reply:
I was trying to assert that View Submitted Application button exits for `pending` app, since we didnt have any button before. Let me add that and clean-up...

Reply:
Updated Facility tests.